src/routes/contact.js

```diff
@@ -199,12 +204,66 @@ router.post('/send', async (req, res) => {
   return res.status(response.getCode()).send(response.getMessage())
 })
 
+
+router.post('/send/:type/:id', async (req, res) => {
+  const response = new utils.Response()
+
+  try {
+    let entity
+    if (req.params.type.toLowerCase() === 'entity') {
+      entity = await models.Entity.findById(req.params.id)
+    } else if (req.params.type.toLowerCase() === 'contact') {
+      entity = await models.Contact.findOne({
+        where: {
+          id: req.params.id
+        }
+      })
+    }
+
+    if (entity.email !== null) {
+      const primary = entity.email.filter(e => e.isPrimary === 'true').length ? entity.email.filter(e => e.isPrimary === 'true')[0] : entity.email[0]
+      // short-lived temporary token that only lasts one hour
+      const temporaryToken = await utils.getToken(req.params.id, primary.address, 'Entity')
+
+      const e = {
+        email: primary.address,
+        name: entity.name,
+        entityName: entity.name,
+        entityId: entity.id,
+        token: temporaryToken
+      }
+      email.sendContactCheckInEmail(e).then(() => {
+        response.setMessage(`${entity.name} emailed sent.`)
+        response.setCode(200)
+      }, err => {
+        response.setMessage('There was an error: ' + err)
+        response.setCode(400)
```

Comment:
`400` is for a bad request; was there anything about the request that causes this pathway to be hit?

Reply:
I updated this to a 500, since this would only happen if there is bad data in the database.